File: trunk/subprojects/android/hackazon/src/com/ntobjectives/hackazon/model/CustomerAddress.java
package com.ntobjectives.hackazon.model;

import java.util.ArrayList;

/**
 * Created with IntelliJ IDEA by Nick Chervyakov.
 * User: Nikolay Chervyakov
 * Date: 31.10.2014
 * Time: 12:30
 */
public class CustomerAddress implements Cloneable {
    public Integer id;
    public String full_name;
    public String address_line_1;
    public String address_line_2;
    public String city;
    public String region;
    public String zip;
    public String country_id;
    public String phone;
    public Integer customer_id;

    @SuppressWarnings("serial")
    public static class List extends ArrayList<CustomerAddress> {
    }

    public static class CustomerAddressesResponse extends CollectionResponse<List> {
    }

    @Override
    public CustomerAddress clone() throws CloneNotSupportedException {
        CustomerAddress address = (CustomerAddress) super.clone();
        address.full_name = full_name;
        address.address_line_1 = address_line_1;
        address.address_line_2 = address_line_2;
        address.city = city;
        address.region = region;
        address.country_id = country_id;
        address.zip = zip;
        address.phone = phone;
        return address;
    }
}

File: classes/App/Rest/Controller/CustomerAddress.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: Nikolay Chervyakov 
 * Date: 20.08.2014
 * Time: 13:01
 */


namespace App\Rest\Controller;


use App\Rest\Controller;

class CustomerAddress extends Controller
{
    protected $modelName = 'CustomerAddress';

    protected function preloadModel()
    {
        $this->model->where('customer_id', '=', $this->getUser()->id());
        parent::preloadModel();
    }
} 